src/adapter/http/src/smart_protocol/ws_tungstenite_client.rs

```diff
@@ -616,7 +626,15 @@ impl SmartTransferProtocolClient for WsSmartTransferProtocolClient {
 
             // Create destination dataset if not exists
             let dst = if let Some(dst) = dst {
-                dst
+                // Check is incoming seed is different from existing
+                if transfer_options.force_update_if_diverged
+                    && let Some((_, first_incoming_block)) = new_blocks.front()
+                    && let odf::MetadataEvent::Seed(seed_event) = &first_incoming_block.event
+                    && &seed_event.dataset_id != dst.get_id()
+                {
+                    InternalError::bail("Rewriting seed block is restricted")?;
+                }
```

Comment:
Can these conditions be isolated in a common handler? It seems we have largely copied them from
`axum_server_push_protocol.rs`